hw2/02-search.js

```diff
@@ -0,0 +1,42 @@
+import { characters } from './02-data.js';
+
+document.addEventListener("DOMContentLoaded", function () {
+    const searchInput = document.getElementById("userInput");
+    const searchButton = document.getElementById("searchButton");
+    const resultsContainer = document.getElementById("resultsContainer");
+
+    function displayResults(results) {
+        let html = '';
+        
+        if (results.length === 0) {
+            html = '<div class="alert alert-warning">No characters found.</div>';
```

Comment:
The approach to handling no results with a warning message is user-friendly and ensures a good user experience by clearly communicating the state of the search, appreciate it :)